### PR TITLE
Fix MAGE C++ API bugs

### PR DESCRIPTION
**Pull request by antepusic**
[master < Task] PR
- [x] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [ ] Provide the full content or a guide for the final git message

### CODE REVIEW

File: include/mgp.hpp [hunk 1]
```diff
@@ -1220,104 +1220,98 @@ class Value {
...
   /// @brief Constructs a Duration value and takes ownership of the given `duration`.
   /// @note The behavior of accessing `duration` after performing this operation is undefined.
   explicit Value(Duration &&duration) {
     ptr_ = mgp::value_make_duration(duration.ptr_);
-    delete &duration;
     duration.ptr_ = nullptr;
   }
 
   Value(const Value &other) : Value(other.ptr_) {}
+  // Value(const Value &other) : Value(mgp::value_copy(other.ptr_, memory)) {}
```
Comment (jbajic) [inline include/mgp.hpp:1314]: Remove commented part

File: include/mgp.hpp [hunk 2]
```diff
@@ -1923,6 +1917,7 @@ void Graph::DeleteRelationship(const Relationship &relationship) { mgp::graph_de
 
 // Nodes:
 
+// TODO: remove comments if possible
```
Comment (jbajic) [inline include/mgp.hpp:1920]: Reminder to not forget this

File: include/mgp.hpp (hunk 1 above)
Comment (antepusic) [reply include/mgp.hpp:1314]: Thanks! I’ll run unit tests and remove the comments when everything’s working.

File: include/mgp.hpp (hunk 2 above)
Comment (antepusic) [reply include/mgp.hpp:1920]: Thanks! I’ll run unit tests and remove the comments when everything’s working.

**Pull request merged**